plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerEditsBookScriptEvent.java

```diff
@@ -51,76 +49,65 @@ public class PlayerEditsBookScriptEvent extends BukkitScriptEvent implements Lis
     public PlayerEditsBookScriptEvent() {
         registerCouldMatcher("player edits book");
         registerCouldMatcher("player signs book");
+        this.<PlayerEditsBookScriptEvent>registerTextDetermination("not_signing", (evt) -> {
+            evt.event.setSigning(false);
+        });
+        this.<PlayerEditsBookScriptEvent, ScriptTag>registerDetermination(null, ScriptTag.class, (evt, context, value) -> {

```

Comment:
This should probably be an optional determination (and return false if the script isn't a book script).